components/MintControls/index.tsx

```diff
@@ -101,16 +123,15 @@ export default function MintControls({ pageState, setPageState }: MintControlsPr
         setTxHash(transaction.hash)
         await transaction.wait()
         const filter = valeNFTinesContract.filters.Transfer(null, recipient)
-        valeNFTinesContract.once(filter, (_from, _to, id) => {
-          setValentineId(id.toString())
-          setPageState(PAGE_STATE.COMPLETE)
-        })
+        const [log] = await valeNFTinesContract.queryFilter(filter, -1, 'latest')
+        setValentineId(log.args.id.toString())
+        setPageState(PAGE_STATE.COMPLETE)
```

Comment:
this is what fixes the opensea issue

Reply:
@JFrankfurt will this look for all events ever that match this? Might link to a past mint?